code/controllers/subsystem/persistence.dm

```diff
@@ -531,19 +535,37 @@ SUBSYSTEM_DEF(persistence)
 
 	WRITE_FILE(file, json_encode(data))
 
-/// Location where we save the information about how many rounds it has been since the engine blew up
+/// Location where we save the information about how many rounds it has been since the engine blew up/tram hits
 #define DELAMINATION_COUNT_FILEPATH "data/rounds_since_delamination.txt"
+#define DELAMINATION_HIGHSCORE_FILEPATH "data/highscore_since_delamination.txt"
+#define TRAM_COUNT_FILEPATH "data/tram_hits_last_round.txt"
 
 /datum/controller/subsystem/persistence/proc/load_delamination_counter()
 	if (!fexists(DELAMINATION_COUNT_FILEPATH))
 		return
 	rounds_since_engine_exploded = text2num(file2text(DELAMINATION_COUNT_FILEPATH))
-	for (var/obj/structure/sign/delamination_counter/sign as anything in GLOB.map_delamination_counters)
-		sign.update_count(rounds_since_engine_exploded)
+	highscore_since_engine_exploded = text2num(file2text(DELAMINATION_HIGHSCORE_FILEPATH))
+	if (fexists(DELAMINATION_HIGHSCORE_FILEPATH))
+		highscore_since_engine_exploded = text2num(file2text(DELAMINATION_HIGHSCORE_FILEPATH))
+	for (var/obj/machinery/incident_display/sign as anything in GLOB.map_delamination_counters)
+		sign.update_delam_count(rounds_since_engine_exploded, highscore_since_engine_exploded)
 
 /datum/controller/subsystem/persistence/proc/save_delamination_counter()
 	rustg_file_write("[rounds_since_engine_exploded + 1]", DELAMINATION_COUNT_FILEPATH)
+	if((rounds_since_engine_exploded + 1) > highscore_since_engine_exploded)
+		rustg_file_write("[rounds_since_engine_exploded + 1]", DELAMINATION_HIGHSCORE_FILEPATH)
+
+/datum/controller/subsystem/persistence/proc/load_tram_counter()
+	if(!fexists(TRAM_COUNT_FILEPATH))
+		return
+	tram_hits_last_round = text2num(file2text(TRAM_COUNT_FILEPATH))
+
+/datum/controller/subsystem/persistence/proc/save_tram_counter()
+	if(SSmapping.config?.map_name == "Tramstation" || SSmapping.config?.map_name == "Birdshot Station")
+		rustg_file_write("[tram_hits_this_round]", TRAM_COUNT_FILEPATH)
```

Comment:
you should not be hardcoding map names into the code like this. Consider using a map trait instead and adding it to the map json files so that tram persistency is tracked.

Reply:
Thanks, I knew there was a better way.